3-create_ros_2_package/README.md

```diff
@@ -0,0 +1,111 @@
+# Exercises 3 - Create ROS 2 package
+For the upcoming exercises, you will start writing some Python code.
+ROS 2 organizes code neatly in **packages** to separate different 
+functionalities, and make them to be easily installable and shareable.
+
+You can follow this tutorial to easily create a new package using 
+[Turtle Nest](https://github.com/Jannkar/turtle_nest) GUI tool, or learn how to create packages manually from the command line by following 
+[the official ROS 2 documentation](https://docs.ros.org/en/humble/Tutorials/Beginner-Client-Libraries/Creating-Your-First-ROS2-Package.html).
+
+## Create a new package
+1. Bring up the Docker container and open a new terminal inside it ([How to run Docker container](/0-setup/Docker%20Cheat%20Sheet.md))
+1. Run a command `turtle-nest` to bring up the guided package creation
+
+Use the following settings to create the package:
+
+**Package name:** `ros2_exercises`
+
+**Destination:** `/home/user/exercises_ws/src`
+
+**Package Type:** `Python`. Uncheck the C++ option
+
+**Python Node Name:** `odometry_publisher` (Will be used for the next exercises)
+
+**Maintainer Name:** `<Your name>`
+
+**Package License:** `No license`
+
+
+Other fields can be left empty / unchecked, as they are not needed in these exercises.
+
+
+<img src="images/turtle_nest_1.png" alt="Turtle Nest 1" width="600">
+<img src="images/turtle_nest_2.png" alt="Turtle Nest 2" width="600">
+<img src="images/turtle_nest_3.png" alt="Turtle Nest 3" width="600">
+
+
+
+## Build and source the package
+
+Next, you will need to compile your code by building the package and then source it for ROS 2 to be able to find it.
+Run the following commands:
+
+    cd /home/user/exercises_ws/
+    colcon build --symlink-install
+    source install/setup.bash
+
+This will build and source all the packages inside your exercises_ws -workspace. 
+We use the `--symlink-install` to make our Python code symbolically linked, so that we don't have to rebuild every time we modify our code.
+
+**NOTE**: You need to do this build and source step every time you relaunch your Docker container!
+
+## What our new package includes?
+
+Let's check what our newly created package contains.
+
+We created our package in the `exercises_ws` folder. 
+This folder mounted by our docker-compose file, meaning that we can access it on our host Ubuntu system as well in `$HOME/exercises_ws`, 
+and all the code you write there will be automatically synchronized inside the container.
+
+Open the exercises_ws in your favorite IDE to take a closer look at what the new package contains.
+For this course, we'll mainly use [PyCharm (Community Edition)](https://www.jetbrains.com/pycharm/download/?section=linux) as the IDE, 
```

Comment:
I changed the wording so that these examples are only done with PyCharm. So that they can really use anything. We can recommend VSCode for more lightweight option with VMs. But I think that VM users should probably run the IDE on the host machine and then just move the code between the VM and host.

Reply:
this is still going to be complicated...they can use whatever they want actually, but i guess just to follow they can also use just the regular text editor...we just need to agree on what we tell them tomorrow

Reply:
adding a reminder to discuss about this in the morning. but for now this can be resolved